src/components/project/project.service.ts

```diff
@@ -627,42 +627,23 @@ export class ProjectService {
       input,
       userId: session.userId,
     });
-    //get a list of engagements
-    const listQuery = this.db.query().call(matchRequestingUser, session);
-    listQuery.match([
-      node('project', 'Project', { active: true, id: project.id }),
-      relation('out', '', 'engagement', { active: true }),
-      node('engagement', 'BaseNode', { active: true }),
-    ]);
-    listQuery.optionalMatch([...this.propMatch('engagement')]).returnDistinct([
+
+    const result = await this.engagementService.list(
       {
-        canReadEngagement: [{ read: 'canRead' }],
-        engagement: [{ id: 'id' }],
+        ...input,
+        filter: {
+          ...input.filter,
+          projectId: project.id,
+        },
       },
-    ]);
-
-    let result;
-    try {
-      result = await listQuery.run();
-    } catch (e) {
-      this.logger.error('e :>> ', e);
-    }
-
-    const items = result
-      ? await Promise.all(
-          result.map((r) => this.engagementService.readOne(r.id, session))
-        )
-      : [];
+      session
+    );
 
-    const retVal: SecuredEngagementList = {
-      total: items.length,
-      hasMore: false,
-      items,
-      canRead: true,
+    return {
+      ...result,
       canCreate: true,
+      canRead: true,
```

Comment:
You'll need to return the user's permissions on the `engagement` property. Looks like you may need to query readOne to get them. 

One of the reasons for these refactors is to implement all the permissioning concepts. Whenever you see constant return values you'll need to refactor them to get the actual values from the db.

Reply:
Got it.

Reply:
Just updated it.